### ISSUE TITLE
[Feature]: Improve usefullness of outlier removal for auto-scaling.

### ISSUE THREAD

**Issue by mcourteaux** [wandb/wandb#3855]
### Description

Right now, it does a reasonable job in finding a range where most data sits (using quantiles, I assume), but always drops a little bit of useful data around it. The reason the current approach fails is because if you have NO outliers, it still considers for example quatiles 5% and 95% for the auto range boundaries, which just assumes 10% of your data is outliers. This is obviously wrong, as we assumed no outliers (eg: signal is a sin(x) function).

### Suggested Solution

Use the range like it's implemented right now, and extend it by 5% or 10% in both directions. More specifically:

```py
range =   # ... the current implementation with outlier removal...
range_size = range.max - range.min
range.min -= range_size * 0.05
range.max += range_size * 0.05
```

Now consider my no-outliers example: sin(x). The current auto-range would yield -0.95 to +0.95. The length of that interval is 1.9. Add now 5% of that (which is 0.095) on both sides, and the final range becomes: -1.145 to +1.145. There you go: the sin(x) function is now nicely displayed with a little margin on the both sides.

**Comment by lesliewandb** [comment]
Hi @mcourteaux, thank you for bringing this to our attention. Are you running into this when clicking on the "Remove Outliers in Chart Scaling" button or when using our scikit-learn integration?

**Comment by mcourteaux** [comment]
Yes, I'm talking about the "Remove Outliers in Chart Scaling" button. I have another way to think of this: remove outliers assumes you have outliers, always. Instead, I think it is more useful that it tries to only remove outliers if you have any. This feature is useful, but when you filter down on runs that don't have outliers, the range it finds becomes not useful as it cuts off useful data.

**Comment by mcourteaux** [comment]
Example: this range is useless:
![image](https://user-images.githubusercontent.com/845012/178134655-3fa9922f-3264-40da-a3a3-deb574d68c83.png)
So, I enable outlier removal for chart scaling:
![image](https://user-images.githubusercontent.com/845012/178134678-d0dc9412-8c35-4b69-b3f8-73a0091f81f1.png)
Now it zooms in to the more useful range, but always cuts away the most recent values, as those are highest and considered "outliers".

**Comment by lesliewandb** [comment]
Hi @mcourteaux, thank you so much for the images they help a lot in understanding the feature request. I'll create a ticket for this and let you know if there are any updates from our engineering team.

**Comment by mcourteaux** [comment]
Any update on this? This is literally a two minute fix. :disappointed:
It's literally this (Python pseudocode):
```py
range = np.quantile(data, [0.05, 0.95])
range_size = range.max - range.min
range.min -= range_size * 0.1
range.max += range_size * 0.1
display_chart(data, y_range=range)
```

**Comment by lesliewandb** [comment]
Hi @mcourteaux, thank you for being patient with us here. I have further escalated this issue to our team, but there has to be a discussion on how this is going to change other people's charts.

**Comment by mcourteaux** [comment]
It will make charts more useful, as under most circumstances, this just show the end of most learning curves. Right now, charts showing the loss-function are always cut off towards the end, as I've shown earlier. Basically, it zooms out just a little bit after removing outliers, to include again the stuff that actually aren't outliers.

**Comment by lesliewandb** [comment]
I see, I'll bring this up with our team and I'll let you know if I hear back from them.

**Comment by ArtsiomWB** [comment]
Hey guys, I was notified that this behavior should now be fixed using `Full fidelity` mode in charts. Let us know if you are seeing something else on your end
